src/Microsoft.Identity.Web/WebAppAuthenticationBuilderExtensions.cs

```diff
@@ -196,5 +190,76 @@ public static partial class WebAppAuthenticationBuilderExtensions
 
             return builder;
         }
+
+        internal static void PopulateOpenIdOptionsFromMicrosoftIdentityOptions(OpenIdConnectOptions options, MicrosoftIdentityOptions microsoftIdentityOptions)
+        {
+            options.Authority = microsoftIdentityOptions.Authority;
+            options.ClientId = microsoftIdentityOptions.ClientId;
+            options.ClientSecret = microsoftIdentityOptions.ClientSecret;
+            options.Configuration = microsoftIdentityOptions.Configuration;
+            options.ConfigurationManager = microsoftIdentityOptions.ConfigurationManager;
+            options.GetClaimsFromUserInfoEndpoint = microsoftIdentityOptions.GetClaimsFromUserInfoEndpoint;
+            foreach (ClaimAction c in microsoftIdentityOptions.ClaimActions)
+            {
+                options.ClaimActions.Add(c);
+            }
+
+            options.RequireHttpsMetadata = microsoftIdentityOptions.RequireHttpsMetadata;
+            options.MetadataAddress = microsoftIdentityOptions.MetadataAddress;
+            options.Events = microsoftIdentityOptions.Events;
+            options.MaxAge = microsoftIdentityOptions.MaxAge;
+            options.ProtocolValidator = microsoftIdentityOptions.ProtocolValidator;
+            options.SignedOutCallbackPath = microsoftIdentityOptions.SignedOutCallbackPath;
+            options.SignedOutRedirectUri = microsoftIdentityOptions.SignedOutRedirectUri;
+            options.RefreshOnIssuerKeyNotFound = microsoftIdentityOptions.RefreshOnIssuerKeyNotFound;
+            options.AuthenticationMethod = microsoftIdentityOptions.AuthenticationMethod;
+            options.Resource = microsoftIdentityOptions.Resource;
+            options.ResponseMode = microsoftIdentityOptions.ResponseMode;
+            options.ResponseType = microsoftIdentityOptions.ResponseType;
+            options.Prompt = microsoftIdentityOptions.Prompt;
+
+            foreach (string scope in microsoftIdentityOptions.Scope)
+            {
+                options.Scope.Add(scope);
+            }
+
+            options.RemoteSignOutPath = microsoftIdentityOptions.RemoteSignOutPath;
+            options.SignOutScheme = microsoftIdentityOptions.SignOutScheme;
+            options.StateDataFormat = microsoftIdentityOptions.StateDataFormat;
+            options.StringDataFormat = microsoftIdentityOptions.StringDataFormat;
+            options.SecurityTokenValidator = microsoftIdentityOptions.SecurityTokenValidator;
+            options.TokenValidationParameters = microsoftIdentityOptions.TokenValidationParameters;
+            options.UseTokenLifetime = microsoftIdentityOptions.UseTokenLifetime;
+            options.SkipUnrecognizedRequests = microsoftIdentityOptions.SkipUnrecognizedRequests;
+            options.DisableTelemetry = microsoftIdentityOptions.DisableTelemetry;
+            options.NonceCookie = microsoftIdentityOptions.NonceCookie;
+            options.UsePkce = microsoftIdentityOptions.UsePkce;
+#if DOTNET_50
```

Comment:
Fixing this as there were 2 additional properties in .NET 5 (discovered by the unit test)